### PR TITLE
feat: support blob api in pytorch loader

### PR DESCRIPTION
**Pull request by eddyxu**
Support handling Blob data in PyTorch loader

### CODE REVIEW

File: python/python/lance/torch/data.py [hunk 1]
```diff
@@ -234,6 +243,10 @@ def __init__(
         self._to_tensor_fn = to_tensor_fn
         self._hf_converter = None
 
+        self._blob_columns = self._blob_columns()
+        if self._blob_columns:
+            self.with_row_id = True
```
Comment (chebbyChefNEQ) [inline python/python/lance/torch/data.py:248]: Why is this needed?

File: python/python/lance/torch/data.py [hunk 2]
```diff
@@ -45,17 +46,25 @@ def _fsl_to_tensor(arr: pa.FixedSizeListArray, dimension: int) -> torch.Tensor:
...
     ret = {}
 
-    for col in batch.schema.names:
+    cols = (
+        batch.column_names if isinstance(batch, pa.RecordBatch) else list(batch.keys())
+    )
+    for col in cols:
         arr: pa.Array = batch[col]
 
+        if isinstance(arr, list) and arr and isinstance(arr[0], lance.BlobFile):
```
Comment (chebbyChefNEQ) [inline python/python/lance/torch/data.py:63]: Is there a way to check earlier? Like when constructing the loader?

**Review by chebbyChefNEQ — APPROVED** [review]

File: python/python/lance/torch/data.py (hunk 1 above)
Comment (eddyxu) [reply python/python/lance/torch/data.py:248]: we need row id to call `dataset.take_rows()`

File: python/python/lance/torch/data.py (hunk 2 above)
Comment (eddyxu) [reply python/python/lance/torch/data.py:63]: Yes we probably can. One way to do it is passing more parameters, but it makes user-specified `to_tensor_fn` more complicated.
